build.gradle.kts

```diff
@@ -71,107 +73,208 @@ tasks.jar.configure {
-              }
-              "developers" {
-                "developer" {
-                  "id"("loosebazooka")
-                  "name"("Appu Goundan")
-                  "email"("appu@google.com")
-                }
-              }
-            }
+          name.set("App Engine Gradle Plugin")
+          description.set("This Gradle plugin provides tasks to build and deploy Google App Engine applications.")
+        }
+      }
+    }
+    withType<MavenPublication> {
+      // Use the resolved versions in pom.xml
+      // Gradle might have different resolution rules, so we set the versions
+      // that were used in Gradle build/test.
+      versionMapping {
+        usage(Usage.JAVA_RUNTIME) {
+          fromResolutionResult()
+        }
+        usage(Usage.JAVA_API) {
+          fromResolutionOf("runtimeClasspath")
+        }
+      }
+      pom {
+        // <editor-fold defaultstate="collapsed" desc="Default POM values">
+        developers {
+          developer {
+            id.set("loosebazooka")
+            name.set("Appu Goundan")
+            email.set("appu@google.com")
+          }
+        }
+        url.set("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
+        inceptionYear.set("2016")
+        licenses {
+          license {
+            name.set("The Apache License, Version 2.0")
+            url.set("https://www.apache.org/licenses/LICENSE-2.0.txt")
+            comments.set("A business-friendly OSS license")
+            distribution.set("repo")
           }
-        }.writeTo(outputFile)
+        }
+        issueManagement {
+          system.set("GitHub")
+          url.set("https://github.com/GoogleCloudPlatform/app-gradle-plugin/issues")
+        }
+        scm {
+          url.set("https://github.com/GoogleCloudPlatform/app-gradle-plugin")
+          connection.set("scm:https://github.com/GoogleCloudPlatform/app-gradle-plugin.git")
+          developerConnection.set("scm:git://github.com/GoogleCloudPlatform/app-gradle-plugin.git")
+        }
+        // </editor-fold>
       }
     }
   }
 }
 
 
 // for kokoro releases
-tasks.register<Sync>("prepareRelease") {
-  from(tasks.jar)
-  from(tasks.named("sourceJar"))
-  from(tasks.named("javadocJar"))
-  from(tasks.named("writePom"))
-
-  into("${buildDir}/release-artifacts")
-
-  dependsOn(tasks.build)
+val prepareRelease by tasks.registering(Sync::class) {
+  description = "Copy release artifacts to ${buildDir.name}/release-artifacts/"
+  group = "release"
+  dependsOn(tasks.publish)
+  into("$buildDir/release-artifacts")
+  from(tempRepoPath) {
+    include("**/appengine-gradle-plugin/**/*.jar")
+    include("**/appengine-gradle-plugin/**/*.module")
+    include("**/appengine-gradle-plugin/**/*.pom")
+    // Flatten hierarchy
+    eachFile {
+      path = "plugin-artifacts/$name"
+    }
+  }
+  from(tempRepoPath) {
+    include("**/*.gradle.plugin/**/*.pom")
+    // Flatten hierarchy
+    eachFile {
+      path = "plugin-markers/$name"
+    }
```

Comment:
Are you sure this flattening is required to separate the artifacts? They all have unique names.

Reply:
I’ve no idea on what are the requirements